ts-sdk/whirlpool/tests/createPool.test.ts

```diff
@@ -1,5 +1,143 @@
-import { describe } from "vitest";
+import { describe, it, beforeAll } from "vitest";
+import {
+  createSplashPoolInstructions,
+  createConcentratedLiquidityPoolInstructions,
+} from "../src/createPool";
+import {
+  DEFAULT_FUNDER,
+  resetConfiguration,
+  setDefaultFunder,
+  SPLASH_POOL_TICK_SPACING,
+  WHIRLPOOLS_CONFIG_ADDRESS,
+} from "../src/config";
+import { setupMint } from "./utils/token";
+import { orderMints } from "../src/token";
+import { rpc, sendTransaction, signer } from "./utils/mockRpc";
+import { fetchWhirlpool, getTickArrayAddress, getWhirlpoolAddress } from "@orca-so/whirlpools-client";
+import assert from "assert";
+import type { Address, Lamports } from "@solana/web3.js";
+import { lamports } from "@solana/web3.js";
+import { _TICK_ARRAY_SIZE, getFullRangeTickIndexes, getTickArrayStartTickIndex, priceToSqrtPrice, sqrtPriceToTickIndex } from "@orca-so/whirlpools-core";
+import { setupMintTE, setupMintTEFee } from "./utils/tokenExtensions";
 
-describe.skip("Create Pool", () => {
-  // TODO:
+
+
+describe("Create Pool", () => {
+  let mintA: Address;
+  let mintB: Address;
+
+  beforeAll(async () => {
+    const mint1 = await setupMint();
+    const mint2 = await setupMint();
+    [mintA, mintB] = orderMints(mint1, mint2);
+  })
+
+  it("Should throw an error if funder is not set", async () => {
+    setDefaultFunder(DEFAULT_FUNDER);
+    await assert.rejects(
+      createConcentratedLiquidityPoolInstructions(rpc, mintA, mintB, 64, 1),
+      {
+        name: 'AssertionError',
+        message: 'Either supply a funder or set the default funder'
+      }
+    );
+    setDefaultFunder(signer);
+  });
+
+  it("Should throw an error if token mints are not ordered", async () => {
+    await assert.rejects(
+      createConcentratedLiquidityPoolInstructions(rpc, mintB, mintA, 64, 1),
+      {
+        name: 'AssertionError',
+        message: 'Token order needs to be flipped to match the canonical ordering (i.e. sorted on the byte repr. of the mint pubkeys)'
+      }
+    );
+  })
+
+  it("Should create valid splash pool instructions", async () => {
+    const price = 10;
+    const sqrtPrice = priceToSqrtPrice(price, 6, 6)
+
+    const { instructions, poolAddress } =
+      await createSplashPoolInstructions(rpc, mintA, mintB, price);
+    await sendTransaction(instructions);
+
+    const pool = await fetchWhirlpool(rpc, poolAddress);
```

Comment:
I would structure the test like this:
* create splash pool
* fetch maybe pool before
* fetch signer balance before
* assert pool does not exist
* send tx
* fetch maybe pool
* fetch signer blance after
* assert pool exists
* assert sqrt, mintA, mintB, and tick spacing
* assert that estInitialization cost equals difference in signer balance

That way you don't need the `Should estimate initialization costs correctly` test case entirely

Reply:
And then create more tests with similar structure for:
* Splash pool with TE token
* CL pool
* CL pool with TE token
* etc.